amlb/datautils.py

```diff
@@ -39,11 +39,15 @@ def read_csv(path, nrows=None, header=True, index=False, as_data_frame=True, dty
     :param dtype: data type for columns.
     :return: a DataFrame
     """
+    if dtype is not None and timestamp_column is not None and timestamp_column in dtype:
+            del dtype[timestamp_column]
```

Comment:
Avoid outer context manipulation, instead copy dtype to a new object and then delete timestamp_column.